### PR TITLE
Update _toc.yml.in

### PR DESCRIPTION
**Pull request by Rmalavally**
DRAFT - TOC restructure

### CODE REVIEW

File: docs/sphinx/_toc.yml.in
Comment (samjwu) [inline docs/sphinx/_toc.yml.in]:
```suggestion
- caption: Quick-start
  entries:
  - file: reference/install.rst
```

**Review by samjwu — DISMISSED** [review]
The table of contents has a very specific format it accepts

A good example is the ROCm TOC: https://github.com/RadeonOpenCompute/ROCm/blob/develop/docs/sphinx/_toc.yml.in

File: docs/sphinx/_toc.yml.in
Comment (samjwu) [inline docs/sphinx/_toc.yml.in]:
```suggestion
- caption: What is MIOpen?
  entries:
  - file:
```

File: docs/sphinx/_toc.yml.in
Comment (samjwu) [inline docs/sphinx/_toc.yml.in]:
```suggestion
- caption: Quick-start
  entries:
  - file:
```

**Review by Rmalavally — COMMENTED** [review]
Thanks, Sam.

**Comment by bghimireamd** [comment]
LGTM

**Comment by Rmalavally** [comment]
> LGTM

Thanks so much.

File: docs/sphinx/_toc.yml.in
Comment (LisaDelaney) [inline docs/sphinx/_toc.yml.in:7]: this should not be a heading. it should be an intro page that describes the project.

File: docs/sphinx/_toc.yml.in
Comment (LisaDelaney) [inline docs/sphinx/_toc.yml.in:1]: where is the API ref heading? you seem to have removed the entire former API ref section and there is no organization to the existing pages

File: docs/sphinx/_toc.yml.in
Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in:7]: We are still discussing the content for the Intro page. For now, we have reorganized the ToC on what's available and accurate. We have also removed the reference to release notes, as it is obsolete.

File: docs/sphinx/_toc.yml.in
Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in:1]: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/Rmalavally-patch-1/docs/sphinx/_toc.yml.in

Comment (LisaDelaney) [reply docs/sphinx/_toc.yml.in:1]: look at the build--it's not rendering, so you need to fix the code.

Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in:1]: Yep. It's a work in progress. Won't render yet.

File: docs/sphinx/_toc.yml.in
Comment (LisaDelaney) [reply docs/sphinx/_toc.yml.in:7]: then we need to wait until we have a page to link to. It's confusing to have a heading represent a placeholder for a future page. A heading (='caption' in the toc) should only be there to represent pages that fit into that heading.

Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in:7]: That's a decision we must make. Working with the team.

**Comment by LisaDelaney** [comment]
these are no longer in the toc--is this intentional? Building MIOpen for Embedded Systems, Building the driver, Prerequisites

File: docs/sphinx/_toc.yml.in
Comment (LisaDelaney) [reply docs/sphinx/_toc.yml.in:1]: you haven't added any of the Doxygen content (=the API ref)

**Comment by Rmalavally** [comment]
> these are no longer in the toc--is this intentional? Building MIOpen for Embedded Systems, Building the driver, Prerequisites

The restructure is based on their approval. I am not an expert to make this call. Let me finalize the content and get back to the team.

**Comment by LisaDelaney** [comment]
> > these are no longer in the toc--is this intentional? Building MIOpen for Embedded Systems, Building the driver, Prerequisites
> 
> The restructure is based on their approval. I am not an expert to make this call. Let me finalize the content and get back to the team.

We can't publish their documentation without the API ref. The API ref is the most important part.

File: docs/sphinx/_toc.yml.in
Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in:1]: - file: reference/apireference.rst is what I have from them. Let's hold off until the structure is finalized.

**Comment by Rmalavally** [comment]
> > > these are no longer in the toc--is this intentional? Building MIOpen for Embedded Systems, Building the driver, Prerequisites
> > 
> > 
> > The restructure is based on their approval. I am not an expert to make this call. Let me finalize the content and get back to the team.
> 
> We can't publish their documentation without the API ref. The API ref is the most important part.

Sure. I am aware. We used to have MIOpen on RTD before with other docs. Let's not rush without ensuring we are doing it correctly.

File: docs/sphinx/_toc.yml.in
Comment (LisaDelaney) [reply docs/sphinx/_toc.yml.in:7]: you can use the README file to get a little general info for a 'What is ..?' page until the team gets a chance to add to it. this page doesn't need to super long or elaborate, just has to describe what the project is/does.

Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in:7]: Thanks, Lisa. Thanks so much for your help and comments.

**Comment by LisaDelaney** [comment]
the landing page (index) is not ready. This PR needs more than toc edits before it can be merged...(at a minimum, add the updated index and what is pages)

**Comment by Rmalavally** [comment]
> the landing page (index) is not ready. This PR needs more than toc edits before it can be merged...(at a minimum, add the updated index and what is pages)

I agree. Even with the minimum updates, it may not be consistent. Let's evaluate the implications of making this publicly available for this release.

File: docs/sphinx/_toc.yml.in
Comment (LisaDelaney) [reply docs/sphinx/_toc.yml.in:1]: Doxygen builds content automatically from the header files (which contains the code/text/content for the API). The team does does not supply the API textual content. This content is build into the "Doxygen" folder--this is what you need to include in the toc in order to incorporate the API ref.

File: docs/sphinx/_toc.yml.in
Comment (samjwu) [inline docs/sphinx/_toc.yml.in]: the files here in the TOC are relative to the `docs/` folder

for this to render correctly, `apireference.rst` needs to go into a folder named `reference`

Comment (samjwu) [reply docs/sphinx/_toc.yml.in]: same goes for tutorials

**Comment by Rmalavally** [comment]
@samjwu Using 'caption' breaks the build. Without the use of 'caption', it builds fine. Please advise.

File: docs/tutorials/what-is-MIOpen.rst
Comment (JehandadKhan) [inline docs/tutorials/what-is-MIOpen.rst:4]: Please remove the reference.

File: docs/tutorials/what-is-MIOpen.rst
Comment (JehandadKhan) [inline docs/tutorials/what-is-MIOpen.rst]:
```suggestion
Networks (RNNs), used in both training and inference.
```

**Review by JehandadKhan — CHANGES_REQUESTED** [review]
Minor edits

**Comment by samjwu** [comment]
> @samjwu Using 'caption' breaks the build. Without the use of 'caption', it builds fine. Please advise.

Please refer to this version of the toc. It has the correct formatting to build the documentation

https://github.com/ROCm/MIOpen/blob/cac7a8872707c117d88cb56993470c7f371809cf/docs/sphinx/_toc.yml.in

File: docs/tutorials/what-is-MIOpen.rst
Comment (Rmalavally) [reply docs/tutorials/what-is-MIOpen.rst:4]: I did. Let me resend the PR. Here's the final text,

MIOpen is AMD’s deep learning primitives library, which provides highly optimized and hand-tuned implementations of different operators such as convolution, batch normalization, pooling, softmax, activation and layers for Recurrent Neural Networks (RNNs), used in both training and inference. Moreover, MIOpen is fully open source, including all its GPU kernels, complementing AMD’s open source ROCm stack. MIOpen is the first to extend the open source advantage into GPU vendor libraries thereby, continuing to embark on the same ethos as the deep learning community.

**Comment by Rmalavally** [comment]
> > @samjwu Using 'caption' breaks the build. Without the use of 'caption', it builds fine. Please advise.
> 
> Please refer to this version of the toc. It has the correct formatting to build the documentation
> 
> https://github.com/ROCm/MIOpen/blob/cac7a8872707c117d88cb56993470c7f371809cf/docs/sphinx/_toc.yml.in

If I use hipSPARSElt as the template, then a title is required for each file. The updated ToC with titles builds fine as long as caption is not used. 

https://github.com/ROCm/hipSPARSELt/blob/develop/docs/sphinx/_toc.yml.in

**Comment by LisaDelaney** [comment]
> > > @samjwu Using 'caption' breaks the build. Without the use of 'caption', it builds fine. Please advise.
> > 
> > 
> > Please refer to this version of the toc. It has the correct formatting to build the documentation
> > https://github.com/ROCm/MIOpen/blob/cac7a8872707c117d88cb56993470c7f371809cf/docs/sphinx/_toc.yml.in
> 
> If I use hipSPARSElt as the template, then a title is required for each file. The updated ToC with titles builds fine as long as caption is not used.
> 
> https://github.com/ROCm/hipSPARSELt/blob/develop/docs/sphinx/_toc.yml.in

A title is not required...but I like to include it because, if you don't, it uses the title (H1) from within the file, which is often too long for a toc. You can find more info on correct syntax here: https://sphinx-external-toc.readthedocs.io/en/latest/user_guide/sphinx.html#basic-structure. Note that YAML is very particular about spacing/tabs.

File: docs/sphinx/_toc.yml.in
Comment (LisaDelaney) [inline docs/sphinx/_toc.yml.in:44]: You're referencing a lot of files/folders that don't exit...and need to include the API ref from the doxygen folder (see my previous comment).

Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in:44]: Bibek and I added the API ref from the Doxygen folder. @bghimireamd Please advise.

Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in:44]: Exit? or Exist? We have reused all existing and original files. Sam recommended we create the appropriate folders, and as suggested, we used hipSPARSElt as the template. Should we remove the titles and keep it simple for 6.0?

**Comment by LisaDelaney** [comment]
@Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.

**Comment by Rmalavally** [comment]
> @Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.

Let me check. Could you also update the doc guidelines accordingly? For now, we have, 

>work on splitting up the individual files as needed (if applicable) and/or creating any new files you want (e.g. the "What is X?" page)
create any folders you need and put the appropriate files in the appropriate folders
get the toc file set up the way you want
update the index file(s) to align with the toc structure

**Comment by Rmalavally** [comment]
> > @Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.
> 
> Let me check. Could you also update the doc guidelines accordingly? For now, we have,
> 
> > work on splitting up the individual files as needed (if applicable) and/or creating any new files you want (e.g. the "What is X?" page)
> > create any folders you need and put the appropriate files in the appropriate folders
> > get the toc file set up the way you want
> > update the index file(s) to align with the toc structure

The ROCm .gitignore file has _toc.yml. Assume we will have _toc.yml.in in the .gitignore file. Please confirm.

**Comment by LisaDelaney** [comment]
> > > @Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.
> > 
> > 
> > Let me check. Could you also update the doc guidelines accordingly? For now, we have,
> > > work on splitting up the individual files as needed (if applicable) and/or creating any new files you want (e.g. the "What is X?" page)
> > > create any folders you need and put the appropriate files in the appropriate folders
> > > get the toc file set up the way you want
> > > update the index file(s) to align with the toc structure
> 
> The ROCm .gitignore file has _toc.yml. Assume we will have _toc.yml.in in the .gitignore file. Please confirm.

No, _toc.yml is an autogenerated file (a copy of _toc.yml.in);  _toc.yml.in should not be in the gitignore file.

**Comment by Rmalavally** [comment]
> > > > @Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.
> > > 
> > > 
> > > Let me check. Could you also update the doc guidelines accordingly? For now, we have,
> > > > work on splitting up the individual files as needed (if applicable) and/or creating any new files you want (e.g. the "What is X?" page)
> > > > create any folders you need and put the appropriate files in the appropriate folders
> > > > get the toc file set up the way you want
> > > > update the index file(s) to align with the toc structure
> > 
> > 
> > The ROCm .gitignore file has _toc.yml. Assume we will have _toc.yml.in in the .gitignore file. Please confirm.
> 
> No, _toc.yml is an autogenerated file (a copy of _toc.yml.in); _toc.yml.in should not be in the gitignore file.

Ok. Good to know. Thanks for clarifying.

File: docs/sphinx/_toc.yml.in
Comment (LisaDelaney) [reply docs/sphinx/_toc.yml.in:44]: that was a typo: I meant exist.

**Comment by junliume** [comment]
@Rmalavally this PR is still in draft state, if we are pursuing it to merging, please update it and resolve all the conflicts above. Thanks!

File: .readthedocs.yaml
Comment (LisaDelaney) [inline .readthedocs.yaml:16]: @samjwu is this change correct?

**Review by samjwu — CHANGES_REQUESTED** [review]
`.readthedocs.yaml` should not need to be changed. It is for the ReadtheDocs site build configurations

**Pull request closed without merging**
